src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -569,6 +569,11 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP102",
                 "The \"createObject\" function is not supported. Construct an object literal using {}.");
+
+            public ErrorDiagnostic DoubleQuoteToken() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP103",
+                "Strings are defined using single quotes in bicep.");

```

Comment:
We don't know that they are trying to declare a string btw. Can you change themessage to something like this:
```csharp
$"The following token is not recognized: \"{token}\". Strings are defined using single quotes in bicep."
```